pkg/proxy/service_port_name.go

```diff
@@ -18,21 +18,10 @@ package proxy
 
 import (
 	"fmt"
-	"net"
 
-	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/types"
 )
 
-// LoadBalancer is an interface for distributing incoming requests to service endpoints.
-type LoadBalancer interface {
-	// NextEndpoint returns the endpoint to handle a request for the given
-	// service-port and source address.
-	NextEndpoint(service ServicePortName, srcAddr net.Addr) (string, error)
-	NewService(service ServicePortName, sessionAffinityType api.ServiceAffinity, stickyMaxAgeMinutes int) error
-	CleanupStaleStickySessions(service ServicePortName)
-}
-
 // ServicePortName carries a namespace + name + portname.  This is the unique
 // identfier for a load-balanced service.
 type ServicePortName struct {
```

Comment:
We probably just want types.go, this was in loadbalancer.go which is now userspace.
Do we also want to move ProxyProvider to a merged types.go?